Shape_detection/include/CGAL/Shape_detection/Region_growing/Polygon_mesh/Plane_face_region.h

```diff
@@ -0,0 +1,460 @@
+          h = next(h, pmesh);
+        } while (h != s);
+
+        std::vector<Triangle_3> face_triangulation;
+        internal::triangulate_face<GeomTraits>(pts, face_triangulation);
+        put(m_face_triangulations, i, face_triangulation);
+      }
+
+      CGAL_precondition(faces(m_face_graph).size() > 0);
+      const FT max_distance = parameters::choose_parameter(
+        parameters::get_parameter(np, internal_np::maximum_distance), FT(1));
+      CGAL_precondition(max_distance >= FT(0));
+      m_distance_threshold = max_distance;
+
+      const FT max_angle = parameters::choose_parameter(
+        parameters::get_parameter(np, internal_np::maximum_angle), FT(25));
+      CGAL_precondition(max_angle >= FT(0) && max_angle <= FT(90));
+
+      m_min_region_size = parameters::choose_parameter(
+        parameters::get_parameter(np, internal_np::minimum_region_size), 1);
+      CGAL_precondition(m_min_region_size > 0);
+
+      const FT default_cos_value = static_cast<FT>(std::cos(CGAL::to_double(
+        (max_angle * static_cast<FT>(CGAL_PI)) / FT(180))));
+      const FT cos_value = parameters::choose_parameter(
+        parameters::get_parameter(np, internal_np::cosine_of_maximum_angle), default_cos_value);
+      CGAL_precondition(cos_value >= FT(0) && cos_value <= FT(1));
+      m_cos_value_threshold = cos_value;
+    }
+
+    /// @}
+
+    /// \name Access
+    /// @{
+
+    /*!
+      \brief implements `RegionType::region_index_map()`.
+
+      This function creates an empty property map that maps each face to a std::size_t
+    */
+    Region_index_map region_index_map() {
+      return get(CGAL::dynamic_face_property_t<std::size_t>(), m_face_graph);
+    }
+
+    /*!
+      \brief implements `RegionType::primitive()`.
+
+      This function provides the last primitive that has been fitted with the region.
+
+      \return Primitive parameters that fits the region.
+
+      \pre `successful fitted primitive via successful call of update(region) with a sufficient large region`
+    */
+
+    // TODO: we probably want to return the m_seed_face instead
+    Primitive primitive() const {
+      return m_plane_of_best_fit;
+    }
+
+    /*!
+      \brief implements `RegionType::is_part_of_region()`.
+
+      This function controls if the face `query` is within
+      the `maximum_distance` from the corresponding plane and if the angle between
+      its normal and the plane's normal is within the `maximum_angle`. If both conditions
+      are satisfied, it returns `true`, otherwise `false`.
+
+      \param query
+      `Item` of the query face
+
+      The last parameter is not used in this implementation.
+
+      \return Boolean `true` or `false`
+
+      \pre `query` is a valid const_iterator of `input_range`
+    */
+    bool is_part_of_region(
+      const Item query,
+      const Region&) const
+    {
+      //TODO: that's a bit dummy that we retest points that are already in the region...
```

Comment:
If performance is an issue, we can add a flag that is set when the region needs to be rechecked. But this region type should be pretty fast already as there is no refitting.